src/eventDispatchers/mouseEventHandlers/mouseDown.js

```diff
@@ -29,31 +33,133 @@ export default function (evt) {
 
   let tools;
   const eventData = evt.detail;
-  const element = eventData.element;
-  const coords = eventData.currentPoints.canvas;
-  const distance = 6;
+  const element = evt.detail.element;
+  const coords = evt.detail.currentPoints.canvas;
 
-  // TODO: instead of filtering these for every interaction, we can change our
-  // State's structure to always know these values.
-  // Filter out disabled and enabled
+  // High level filtering
   tools = getInteractiveToolsForElement(element, getters.mouseTools());
-  // Filter out tools that do not match mouseButtonMask
   tools = tools.filter((tool) =>
     isMouseButtonEnabled(eventData.which, tool.options.mouseButtonMask)
   );
-  tools = getToolsWithDataForElement(element, tools);
 
-  // Find tools with handles we can move
-  const toolsWithMoveableHandles = tools.filter((tool) => {
-    const toolState = getToolState(eventData.element, tool.name);
+  const activeTools = tools.filter((tool) => tool.mode === 'active');
+
+  // If any tools are active, check if they have a special reason for dealing with the event.
+  if (activeTools.length > 0) {
+    // TODO: If length > 1, you could assess fitness and select the ideal tool
+    // TODO: But because we're locking this to 'active' tools, that should rarely be an issue
+    const firstActiveTool = activeTools[0];
+
+    if (typeof firstActiveTool.activeMouseDownCallback === 'function') {
+      const shouldPreventPropagation = firstActiveTool.activeMouseDownCallback(evt);
+
+      if (shouldPreventPropagation) {
+        preventPropagation(evt);
+
+        return;
+      }
```

Comment:
I think we may need to separate these two bits of logic.

if `firstActiveTool.activeMouseDownCallback(evt);` returns true, then we should `return;` so the dispatcher doesn't dispatch the event multiple times; however, the callback itself is handed the event, so it can choose to stopPropagation if it would like to?

I don't want to lock users into stopping propagation if they want the eventDispatcher to exit.

(I hope that makes sense)

Reply:
I think this is the only "blocker" for a merge. Can we update this so that:

```
const claimEvent = firstActiveTool.activeMouseDownCallback(evt);
if(claimEvent){ return; }
``` 

And `activeMouseDownCallback(evt)` can call

`evt.stopImmediatePropagation();` and other actions on the event if it wants to?

Reply:
We can talk about this more if you'd like; that's just the preference that gives me the good feels; but I can be swayed :P

Reply:
Yeah, that seems fine to me, simple actually!